docs/design/storage-node-downtime-tracking.md

```diff
@@ -0,0 +1,209 @@
+# Storage Node Downtime Tracking
+
+## Abstract
+
+This document describes storage node downtime tracking.
+
+## Background
+
+[Disqualification design document](disqualification.md) describes how storage nodes get disqualified based on the reputation scores described in [node selection design document](node-selection.md).
+
+Current disqualification based on uptime disqualified several nodes without clear and fair evidence. These disqualifications needed to be reverted and the uptime based disqualification disabled. Before we can start handling disqualifications we need to more reliably track offline status of nodes.
+
+[Kademlia removal design document](kademia-removal.md) describes that each node will need to contact each satellite regularly every hour. This is used in the following design.
+
+This document does not describe how the downtime affects reputation and how disqualifications will work.
+
+## Design
+
+For tracking offline duration we need:
+
+- A new SQL database table to store offline duration.
+- Detecting which nodes are offline.
+- Estimate how long they are offline.
+
+An _uptime check_ referenced in this section is a connection initiated by the satellite to any storage node in the same way that's described in [network refreshing section of the Kademlia removal design document](kademia-removal.md#network-refreshing).
+
+__NOTE__ the SQL code in this section is illustrative for explaining the algorithm concisely.
+
+### Database
+
+The following new SQL database table will store storage nodes offline time.
+
+```sql
+CREATE TABLE nodes_offline_time (
+    node_id    BYTEA NOT NULL,
+    tracked_at timestamp with time zone NOT NULL,
+    seconds    integer -- Measured number of seconds being offline
+)
+```
+
+The current Satellite database has the table `nodes`. For the offline time calculation we use the following columns:
+
+- `last_contact_success`
+- `last_contact_failure`
+- `total_uptime_count`
+- `uptime_success_count`
+
+### Detecting offline nodes
+
+Per [Kademlia removal design document](https://github.com/storj/storj/blob/master/docs/design/kademlia-removal.md#network-refreshing), any storage node has to ping the satellite every hour. For storage nodes that have not pinged, we need to contact them directly.
+
+For finding the potentially offline storage nodes, we run a chore, with the following query:
```

Comment:
For finding the storage nodes gone offline, we run a chore, with the following query: